src/service.rs

```diff
@@ -59,23 +64,21 @@ where
         Box::pin(async move {
             let config = store.config.clone();
             let mut cookies = get_cookies(&req);
+            let mut is_new_id = false;
 
             let session = AxumSession {
                 id: {
-                    let id = if let Some(cookie) = cookies.get(&store.config.cookie_name) {
-                        (
-                            AxumSessionID(match Uuid::parse_str(cookie.value()) {
-                                Ok(v) => v,
-                                Err(_) => {
-                                    tracing::warn!("Uuid \" {} \" is invalid", cookie.value());
-                                    return Ok(Response::builder()
-                                        .status(StatusCode::UNAUTHORIZED)
-                                        .body(body::boxed(Full::from("401 Unauthorized")))
-                                        .unwrap());
-                                }
-                            }),
-                            true,
-                        )
+                    if let Some(cookie) = cookies.get(&store.config.cookie_name) {
+                        AxumSessionID(match Uuid::parse_str(cookie.value()) {
```

Comment:
You've taken this "everything is an expression" a bit too far. Just going from the inside you've got return inside a match inside a ctor inside an if let inside a ctor!

Reply:
please recheck this!